CRM/Activity/Import/Parser/Activity.php

```diff
@@ -436,4 +406,62 @@ protected function deprecated_activity_formatted_param(&$params, &$values, $crea
     return NULL;
   }
 
+  /**
+   *
+   * Get the value for the given field from the row of values.
+   *
+   * @param array $row
+   * @param string $fieldName
+   *
+   * @return null|string
+   */
+  protected function getFieldValue(array $row, string $fieldName) {
+    return $row[$this->getFieldIndex($fieldName)] ?? NULL;
+  }
+
+  /**
+   * Get the index for the given field.
+   *
+   * @param string $fieldName
+   *
+   * @return false|int
+   */
+  protected function getFieldIndex(string $fieldName) {
+    return array_search($fieldName, $this->_mapperKeys, TRUE);
+
+  }
+
+  /**
+   * Add an error to the values.
+   *
+   * @param array $values
+   * @param string $error
+   *
+   * @return int
+   */
+  protected function addError(array &$values, string $error): int {
+    array_unshift($values, $error);
+    return CRM_Import_Parser::ERROR;
+  }
+
+  /**
+   * Validate that the activity type id does not conflict with the label.
+   *
+   * @param array $values
+   *
+   * @return void
+   * @throws \CRM_Core_Exception
+   */
+  protected function validateActivityTypeIDAndLabel(array $values): void {
+    $activityLabel = $this->getFieldValue($values, 'activity_label');
+    $activityTypeID = $this->getFieldValue($values, 'activity_type_id');
+    if ($activityLabel && $activityTypeID
+      && $activityLabel !== CRM_Core_PseudoConstant::getLabel('CRM_Activity_BAO_Activity', 'activity_type_id', $activityTypeID)) {
+      throw new CRM_Core_Exception(ts('Activity label and Activity Type ID are in conflict'));
+    }
+    if (!$activityLabel && !$activityTypeID) {
+      throw new CRM_Core_Exception(ts('Missing required fields') . ' ' . implode(ts(' or '), [ts('Activity label'), ts('Activity Type id')]));
```

Comment:
I did this just because I thought it might avoid adding a new string to be translated

Reply:
I understand what you're saying but most of these strings don't exist currently, and `' or '` might be awkward to translate without context.

Also for this one and the error on 458 it should be "activity type label" not "activity label".

Reply:
Also, as a rule, leading and trailing whitespace is not allowed in strings passed to `ts()`.

Reply:
OK - I've pushed up as a single string